index.js

```diff
@@ -1,46 +1,61 @@
 'use strict';
-// const dataModel = require('screwdriver-data-model');
+const Datastore = require('screwdriver-datastore-base');
+const schemas = require('screwdriver-data-schema');
+const vogels = require('vogels');
 
-/**
- * @constructor
- * @method EngineDynamodbStore
- * @param  {Object}            config Base configuration to be merged with Dynamodb config
- */
-function EngineDynamodbStore(config) {
-    this.config = config;
+const DEFAULT_REGION = 'us-west-2';
 
-    this.client = null;
+class Dynamodb extends Datastore {
+    /**
+     * Constructs a Dynamodb object
+     * @param  {Object} config           Configuration object
+     * @param  {String} config.tableName Name of the DynamoDB table to interact with
+     * @param  {String} [config.region]  AWS region to operate in
+     */
+    constructor(config) {
+        const tableName = config.tableName;
+        const region = config.region || DEFAULT_REGION;
+
+        super();
+        vogels.AWS.config.update({ region });
+
+        this.client = vogels.define(tableName, {
```

Comment:
this client needs to be stored in an Object so that we can share this datastore module among all the tables